sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -159,7 +159,7 @@ class SQLAppStatusListener(
   }
 
   private def aggregateMetrics(exec: LiveExecutionData): Map[Long, String] = {
-    val metricIds = exec.metrics.map(_.accumulatorId).sorted
+    val metricIds = exec.metrics.map(_.accumulatorId).toSet
     val metricTypes = exec.metrics.map { m => (m.accumulatorId, m.metricType) }.toMap
     val metrics = exec.stages.toSeq
       .flatMap { stageId => Option(stageMetrics.get(stageId)) }
```

Comment:
Consider also change the following `flatMap` / `filter`  / `groupBy` into `while` loop

Reply:
not sure what you mean here. Why should we use a `while` loop?

Reply:
If the metrics is large, then using a while loop can reduce the number of traversal loops. And it is not complicated to do it in the code here.

Reply:
I am also fine with the current code here.

Reply:
yes, we can save 1 traversal, but I am not sure it is worth honestly... This approach seems cleaner to me.